fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RelOpValue.java

```diff
@@ -440,6 +450,32 @@ private static Value encapsulate(@Nonnull TypeRepository.Builder typeRepositoryB
         }
     }
 
+    /**
+     * The {@code in} function.
+     */
+    @AutoService(BuiltInFunction.class)
+    public static class InFn extends BuiltInFunction<Value> {
+        public InFn() {
+            super("in",
+                    List.of(new Type.Any(), new Type.Array()), InFn::encapsulate);
+        }
+
+        private static Value encapsulate(@Nonnull TypeRepository.Builder typeRepositoryBuilder, @Nonnull BuiltInFunction<Value> builtInFunction, @Nonnull final List<Typed> arguments) {
+            return RelOpValue.encapsulate(typeRepositoryBuilder.build(), builtInFunction.getFunctionName(), Comparisons.Type.IN, arguments);
+        }
+
+        public static Object evaluate(Object left, Object right) {
+            Verify.verify(right instanceof List);
+            if (((List<?>) right).stream().anyMatch(object -> object != null && object.equals(left))) {
+                return true;
+            } else if (((List<?>) right).stream().anyMatch(Objects::isNull)) {
+                return null;
+            } else {
+                return false;
```

Comment:
This is subtle (in a good way); at first glance I thought it is somewhat incorrect, but it seems to be actually correct.

Reply:
:)